### PR TITLE
micro: port operator DIV kernel from lite with test

### PR DESCRIPTION
**Pull request by ddavis-2015**
Complete implementation of TFLM operator DIV kernel and associated TFLM test code.

This PR represents steps 4 & 5 of the work to port operator DIV as tracked in Issue #45431

### CODE REVIEW

**Review by petewarden — APPROVED** [review]

**Comment by gbaned** [comment]
@ddavis-2015 This PR is in draft, can you please take a look on this? Please. Thanks!

**Review by advaitjain — CHANGES_REQUESTED** [review]
I'm guessing that this PR is in draft because it is not ready for review yet.

It looks like it needs to be rebased once the earlier PRs are merged so that we can see the diff between the file that was copied from lite and the version that will be ported to micro.

File: tensorflow/lite/micro/kernels/div.cc [hunk 1]
```diff
@@ -105,128 +71,132 @@ TfLiteStatus Prepare(TfLiteContext* context, TfLiteNode* node) {
...
+      TF_LITE_DIV(reference_ops, BroadcastDivSlow, uint8_t);
     } else {
-      if (need_broadcast) {
-        TF_LITE_DIV(optimized_ops, BroadcastDivSlow, uint8_t);
-      } else {
-        TF_LITE_DIV(optimized_ops, Div, uint8_t);
-      }
+      TF_LITE_DIV(reference_ops, Div, uint8_t);
     }
-#undef TF_LITE_DIV
```
Comment (petewarden) [inline tensorflow/lite/micro/kernels/div.cc:195]: Any reason this undef was moved? Keeping it in the  smallest scope possible seems mildly preferable.

**Review by petewarden — CHANGES_REQUESTED** [review]

File: tensorflow/lite/micro/kernels/div.cc (hunk 1 above)
Comment (ddavis-2015) [reply tensorflow/lite/micro/kernels/div.cc:195]: Minimal scope is now used for the TF_LITE_DIV definition.

**Comment by gbaned** [comment]
@ddavis-2015  This PR is in draft, any update on this? Please. Thanks!

**Review by petewarden — APPROVED** [review]

**Comment by rthadur** [comment]
@ddavis-2015 seeing this error internally 

`third_party/tensorflow/lite/micro/kernels/div.cc:68:30: error: implicit conversion increases floating-point precision: 'float' to 'const double' [-Werror,-Wdouble-promotion]
        input1->params.scale / (input2->params.scale * output->params.scale);
        ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.`

**Comment by ddavis-2015** [comment]
> @ddavis-2015 seeing this error internally
> 
> `third_party/tensorflow/lite/micro/kernels/div.cc:68:30: error: implicit conversion increases floating-point precision: 'float' to 'const double' [-Werror,-Wdouble-promotion] input1->params.scale / (input2->params.scale * output->params.scale); ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ 1 error generated.`

Thanks for the heads up.  A new commit has been pushed and hopefully there will be no more compile errors.

**Review by rthadur — APPROVED** [review]

**Comment by petewarden** [comment]
It looks like third_party/tensorflow/lite/kernels:div_test is failing:

```
[ RUN      ] QuantizedDivOpTest.QuantizedNoActivationUInt8
*** SIGABRT received by PID 3118 (TID 3118) on cpu 36 from PID 3118; stack trace: ***
PC: @     0x7ff197c5e602  (unknown)  raise
    @     0x7ff18b528bf6        960  FailureSignalHandler()
    @     0x7ff1a126a9a0  (unknown)  (unknown)
    @     0x7ff1f581eb04         80  tflite::reference_ops::DivElementwise<>()
    @     0x7ff1f5823387        304  tflite::ops::builtin::div::EvalQuantized<>()
    @     0x7ff1f581b5b2         80  tflite::ops::builtin::div::Eval<>()
    @     0x7ff1ef10f8e5        160  tflite::Subgraph::Invoke()
    @     0x7ff1ef113d50         96  tflite::Interpreter::Invoke()
    @     0x7ff1f6618db4        112  tflite::SingleOpModel::Invoke()
    @     0x7ff1f86662cb       1936  tflite::(anonymous namespace)::QuantizedDivOpTest_QuantizedNoActivationUInt8_Test::TestBody()
    @     0x7ff1a583bdaa         48  testing::Test::Run()
    @     0x7ff1a583d353         64  testing::TestInfo::Run()
    @     0x7ff1a583e0c0         80  testing::TestSuite::Run()
    @     0x7ff1a585023c        192  testing::internal::UnitTestImpl::RunAllTests()
    @     0x7ff1a584f7f5         64  testing::UnitTest::Run()
    @     0x7ff1f82497cd        112  main
    @     0x7ff197c4abbd        208  __libc_start_main
    @     0x56342f43ce69  (unknown)  ../sysdeps/x86_64/start.S:108 _start
```

You should be able to reproduce this with something like:

`bazel test third_party/tensorflow/lite/kernels:div_test'

**Review by petewarden — APPROVED** [review]

**Pull request merged**